### PR TITLE
add waila support

### PR DESCRIPTION
**Pull request by asdflj**
add waila support

### CODE REVIEW

**Comment by asdflj** [comment]
@GlodBlock

File: src/main/java/com/glodblock/github/crossmod/waila/PartWailaDataProvider.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import java.util.List;
+
+public class PartWailaDataProvider implements IWailaDataProvider {
+    private final List<IPartWailaDataProvider> providers;
+    private final PartAccessor accessor = new PartAccessor();
+    private final Tracer tracer = new Tracer();
+
+    public PartWailaDataProvider() {
+        final IPartWailaDataProvider speed = new SpeedWailaDataProvider();
+        this.providers = Lists.newArrayList(speed);
```
Comment (GlodBlock) [inline src/main/java/com/glodblock/github/crossmod/waila/PartWailaDataProvider.java:29]: use `Collections.singletonList` here if there is only one element

File: src/main/java/com/glodblock/github/crossmod/waila/part/SpeedWailaDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    public List<String> getWailaBody(
+        final IPart part,
+        final List<String> currentToolTip,
+        final IWailaDataAccessor accessor,
+        final IWailaConfigHandler config) {
+        if (part instanceof PartSharedFluidBus) {
+            part.readFromNBT(accessor.getNBTData());
+            currentToolTip.add(String.format("%s: %s mB", I18n.format(NameConst.WAILA_SPEED), Utility.formatNumbers(((PartSharedFluidBus) part).calculateAmountToSend())));
```
Comment (GlodBlock) [inline src/main/java/com/glodblock/github/crossmod/waila/part/SpeedWailaDataProvider.java:28]: mB/t or mB/s ?

**Review by GlodBlock — CHANGES_REQUESTED** [review]

File: src/main/java/com/glodblock/github/crossmod/waila/part/SpeedWailaDataProvider.java (hunk 2 above)
Comment (asdflj) [reply src/main/java/com/glodblock/github/crossmod/waila/part/SpeedWailaDataProvider.java:28]: image

Comment (GlodBlock) [reply src/main/java/com/glodblock/github/crossmod/waila/part/SpeedWailaDataProvider.java:28]: use the 5 ticks as display value

**Comment by GlodBlock** [comment]
and why the slots in dual interface got invisible?

**Comment by asdflj** [comment]
> and why the slots in dual interface got invisible?

7N$05WJH$~~G$QE~FM7WH~A

**Review by GlodBlock — APPROVED** [review]

**Pull request merged**